packages/core/src/agent/openai/worker.ts

```diff
@@ -214,7 +214,19 @@ export class OpenAIAgentWorker
       (r: ChatResponseChunk) => new Response(r.delta),
     );
 
-    return new StreamingAgentChatResponse(newStream, task.extraState.sources);
+    // prepend content from first chunk that was already read
+    const prependedStream = (async function* () {
+      yield new Response(content);
+
+      for await (const response of newStream) {
+        yield response;
+      }
+    })();
```

Comment:
@himself65 that's pretty ugly - anything more elegant?

Reply:
updated

Reply:
https://github.com/run-llama/LlamaIndexTS/pull/706/commits/0bdc6d838a654fa6e038f336a9ee9a0cc75125fa